broadinstitute/single_cell_portal_core
# Enable linking clusters to external resources (SCP-4668)

This lets study editors show a link to an external resource for a cluster, e.g. to a Nissl stain image for a cluster scatter plot.

It's part of our U19 Mouse Atlas dataset work.  Here's how it looks:

https://user-images.githubusercontent.com/1334561/191530408-abba34e8-fa30-42f4-85cd-d40f685883cc.mp4

(The unusual transitions in above video are me cutting out sluggish local wait times.  I used [Canva](https://www.canva.com/features/video-trimmer/).)

To test:
* Go to a study with visualizations
* Go to its upload UI, then "Clustering" tab
* Add values for the new external link fields, e.g.:
  * URL: http://brainarchitecture.org/viewer4/mouse/map/13652N/3483091
  * Title: Brain architecture section 1
  * Description: Corresponding registered brain section (MD699) (section 368) 
* Click "Save"
* In your `rails console -e development`, run `Rails.cache.clear`
* Go to Study page, Explore tab
* Note link, click it, confirm it opens external web resource in new tab

This satisfies SCP-4668.
